### PR TITLE
Added a composer.json to enable support for PIE

### PR DESCRIPTION
**Pull request by asgrim**
Adds support for [PIE](https://github.com/php/pie).

Once merged, a maintainer would need to add this repo's URL to https://packagist.org/packages/submit :)

More info/documentation can be read on: https://github.com/php/pie/blob/main/docs/extension-maintainers.md

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
+
+{
+    "name": "phpredis/phpredis",
```
Comment (asgrim) [inline composer.json:3]: Please verify vendor namespace & package name are acceptable

Comment (michael-grunder) [reply composer.json:3]: Seems right to me, although I need to read up on PIE

**Comment by michael-grunder** [comment]
This is neat. Is PIE intended to replace pecl?

@yatsukhnenko Seems like it should be fine to merge pre-6.1.0RC2

**Comment by asgrim** [comment]
> This is neat. Is PIE intended to replace pecl?
> 
> @yatsukhnenko Seems like it should be fine to merge pre-6.1.0RC2

Yep, the goal is to eventually replace PECL for new extension releases (although PECL will not disappear, otherwise so much stuff will break!) - just trying to onboard a few more extensions at the moment, so we can ensure things will work; it's not quite ready for prime time, but we hope to have a working "0.0.1" release out to coincide with PHP 8.4 release in Nov.

If you do spot any issues when trying things out, please do let me know! :)

**Comment by yatsukhnenko** [comment]
@michael-grunder if I correctly understand we don't need to add this file to pecl package so it's safe to merge it to 6.1.0RC2

**Comment by michael-grunder** [comment]
Merged and submitted, thanks!

Let me know if you need anything else from us.

**Comment by asgrim** [comment]
Thanks, works nicely! :ok_hand: 

```
$ bin/pie install --with-php-config=/home/james/workspace/oss/php-src-build/bin/php-config phpredis/phpredis:dev-develop
This command may need elevated privileges, and may prompt you for your password.
You are running PHP 8.3.11
Target PHP installation: 7.4.33 ts, on Linux/OSX/etc x86_64 (from /home/james/workspace/oss/php-src-build/bin/php)
Found package: phpredis/phpredis:dev-develop which provides ext-redis
phpize complete.
Configure complete with options: --with-php-config=/home/james/workspace/oss/php-src-build/bin/php-config
Build complete: /tmp/pie_downloader_66edbf31e4f998.58310460/phpredis-phpredis-b59e35a/modules/redis.so
Install complete: /home/james/workspace/oss/php-src-build/lib/php/extensions/debug-zts-20190902/redis.so
You must now add "extension=redis" to your php.ini
```

thank you folks!

**Pull request merged**